.github/workflows/ci.yml

```diff
@@ -1,18 +1,87 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+---
 name: CI
 on:
-  push:
+  workflow_dispatch:
```

Comment:
Nit: Do we want to change this triggering behavior? Should CI run on every commit?

Reply:
I don't think there's much benefit to triggering on every push, a run for `pull_request` shows a working test suite when reviewing a PR, and we can run via dispatch on-demand.

Reply:
Maybe push is a little easier for development, though I don't expect this Action to change much going forward other than dependabot.